src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -59,16 +102,58 @@ public T remove(int index) {
 
     @Override
     public boolean remove(T object) {
+        Node<T> searchNode = first;
+        while (searchNode != null) {
+            T item = searchNode.getItem();
+            if (Objects.equals(item, object)) {
+                removeNode(searchNode);
+                return true;
+            }
+            searchNode = searchNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        checkIndex(index);
+        Node searchNode = first;
+        for (int i = 0; i < index; i++) {
+            searchNode = searchNode.next;
+            if (searchNode == null) {
+                return null;
+            }
+        }
+        return searchNode;
+
+    }
+
+    private void removeNode(Node<T> node) {
+        if (node.prev != null) {
```

Comment:
You need to create unlink method and separate this king of logic there, see checklist